pkg/asset/agent/manifests/agentpullsecret.go

```diff
@@ -63,6 +63,9 @@ func (a *AgentPullSecret) Generate(dependencies asset.Parents) error {
 			StringData: map[string]string{
 				pullSecretKey: installConfig.Config.PullSecret,
 			},
+			Data: map[string][]byte{
```

Comment:
This doesn't seem to be required: it's ok to generate a secret filling the `StringData` field only (when the source it's `install-config.yaml`)